internal/core/src/index/skipindex_stats/SkipIndexStats.h

```diff
@@ -0,0 +1,1116 @@
+        }
+        return {total_rows,
+                null_count,
+                min,
+                max,
+                contains_true,
+                contains_false,
+                std::move(unique_values)};
+    }
+
+    metricsInfo<std::string>
+    ProcessStringFieldMetrics(const StringChunk* chunk) const {
+        // all captured by reference
+        bool has_first_valid = false;
+        int64_t total_rows = chunk->RowNums();
+        int64_t null_count = 0;
+        std::string_view min;
+        std::string_view max;
+        ankerl::unordered_dense::set<std::string_view> unique_values;
+        ankerl::unordered_dense::set<std::string_view> ngram_values;
+
+        for (int64_t i = 0; i < total_rows; ++i) {
+            bool is_valid = chunk->isValid(i);
+            if (!is_valid) {
+                null_count++;
+                continue;
+            }
+            auto value = chunk->operator[](i);
+            if (!has_first_valid) {
+                min = value;
+                max = value;
+                has_first_valid = true;
+            } else {
+                if (value < min) {
+                    min = value;
+                }
+                if (value > max) {
+                    max = value;
+                }
+            }
+            if (unique_values.find(value) != unique_values.end()) {
+                continue;
+            }
+            unique_values.insert(value);
+            size_t length = value.length();
+            if (length >= DEFAULT_SKIPINDEX_MIN_NGRAM_LENGTH) {
+                for (size_t j = 0;
+                     j + DEFAULT_SKIPINDEX_MIN_NGRAM_LENGTH <= length;
+                     ++j) {
+                    std::string_view ngram =
+                        value.substr(j, DEFAULT_SKIPINDEX_MIN_NGRAM_LENGTH);
+                    ngram_values.insert(ngram);
+                }
+            }
+        }
+        return {total_rows,
+                null_count,
+                min,
+                max,
+                false,
+                false,
+                std::move(unique_values),
+                std::move(ngram_values)};
+    }
+
+    template <typename T>
+    std::unique_ptr<FieldChunkMetrics>
+    LoadMetrics(const metricsInfo<T>& info) const {
+        if (info.total_rows_ - info.null_count_ == 0) {
+            return nullptr;
+        }
+        if constexpr (std::is_same_v<T, bool>) {
+            return std::make_unique<BooleanFieldChunkMetrics>(
+                info.contains_true_, info.contains_false_);
+        }
+        T min, max;
+        if constexpr (std::is_same_v<T, std::string>) {
+            min = std::string(info.min_);
+            max = std::string(info.max_);
+        } else {
+            min = info.min_;
+            max = info.max_;
+        }
+        if constexpr (std::is_floating_point_v<T>) {
+            return std::make_unique<FloatFieldChunkMetrics<T>>(min, max);
+        }
+        BloomFilterPtr bloom_filter =
```

Comment:
So, we will always build and load bloom filter as well as ngram rather than controlled by configs?